File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第2章分数.py
#lanqiaoOJ 610
#[分数 2018年第九届蓝桥杯C/C++大学A组，Java大学A组 ] 
#代码1：
a,b = 1,0
for i in range(0,20):  
    b += a
    a *= 2    
print("%d/%d"%(b,a/2))

#代码2：
print("%d/%d"%(2 ** 20 -1,2 ** 19))

#代码3：
print(str(2 ** 20 -1)+'/'+str(2 ** 19))

#代码4：
print("%d/%d"% (pow(2,20)-1 ,pow(2,19)))

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第2章分数.cpp
// lanqiaoOJ 610
//[分数 2018年第九届蓝桥杯C/C++大学A组，Java大学A组 ] 


#include<bits/stdc++.h>
using namespace std;
int main(){
	int a = (1 << 20) - 1; //分子
	int b = (1 << 19);     //分母
	int t = __gcd(a, b);   //除去公约数
	cout << a/t << "/" << b/t;
	return 0;
}